ibm/ibm.go

```diff
@@ -0,0 +1,209 @@
+package ibm
+
+import (
+	"fmt"
+	"time"
+
+	bluemix "github.com/IBM-Cloud/bluemix-go"
+	v2 "github.com/IBM-Cloud/bluemix-go/api/container/containerv2"
+	"github.com/IBM-Cloud/bluemix-go/session"
+	"github.com/libopenstorage/cloudops"
+	"github.com/libopenstorage/cloudops/backoff"
+	"github.com/libopenstorage/cloudops/unsupported"
+)
+
+const (
+	labelWorkerPoolName = "ibm-cloud.kubernetes.io/worker-pool-name"
+	labelWorkerPoolID   = "ibm-cloud.kubernetes.io/worker-pool-id"
+	vpcProviderName     = "vpc-gen2"
+)
+
+type ibmOps struct {
+	cloudops.Compute
+	cloudops.Storage
+	ibmClusterClient v2.ContainerServiceAPI
+	inst             *instance
+}
+
+// instance stores the metadata of the running ibm instance
+type instance struct {
+	name            string
+	hostname        string
+	zone            string
+	region          string
+	resourceGroup   string
+	clusterName     string
+	clusterLocation string
+	nodePoolID      string
+}
+
+// NewClient creates a new IBM operations client
+func NewClient() (cloudops.Ops, error) {
+
+	var i = new(instance)
+
+	c := new(bluemix.Config)
+
+	sess, err := session.New(c)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get session. error: [%v]", err)
+	}
+
+	ibmClusterClient, err := v2.New(sess)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get ibm cluster client. error: [%v]", err)
+	}
+
+	instanceName, clusterName, resourceGroup, err := getInfoFromEnv()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get cluster info. error: [%v] ", err)
+	}
+
+	i.name = instanceName
+	i.clusterName = clusterName
+	i.resourceGroup = resourceGroup
+
+	return backoff.NewExponentialBackoffOps(
+		&ibmOps{
+			Compute:          unsupported.NewUnsupportedCompute(),
+			Storage:          unsupported.NewUnsupportedStorage(),
+			ibmClusterClient: ibmClusterClient,
+			inst:             i,
+		},
+		isExponentialError,
+		backoff.DefaultExponentialBackoff,
+	), nil
+}
+
+func (i *ibmOps) Name() string {
+	return string(cloudops.IBM)
+}
+
+func (i *ibmOps) InstanceID() string {
+	return i.inst.name
+}
+
+func isExponentialError(err error) bool {
+	return true
+}
+
+func (i *ibmOps) InspectInstance(instanceID string) (*cloudops.InstanceInfo, error) {
+	target := v2.ClusterTargetHeader{
+		ResourceGroup: i.inst.resourceGroup,
+		Provider:      vpcProviderName,
+	}
+	workerDetails, err := i.ibmClusterClient.Workers().Get(i.inst.clusterName, instanceID, target)
+	if err != nil {
+		return nil, err
+	}
+
+	instanceInfo := &cloudops.InstanceInfo{
+		CloudResourceInfo: cloudops.CloudResourceInfo{
+			Name: workerDetails.ID,
+			Labels: map[string]string{
+				labelWorkerPoolName: workerDetails.PoolName,
+				labelWorkerPoolID:   workerDetails.PoolID,
+			},
+			Zone:   workerDetails.Location,
+			Region: workerDetails.Location,
```

Comment:
should this be workerDetails.Region ?

Reply:
IBM SDK returns location as the only thing for worker details. Hence, populationg same for both zone and region for now.  As of now, there is no dependacy on these fields in test code.